harness/determined/common/api/experiment.py

```diff
@@ -147,7 +151,15 @@ def create_experiment(
         raise Exception(r)
 
     if validate_only:
-        return 0
+        return 0, None
+
+    try:
+        warnings = r.json().get("warnings")
+        if warnings:
+            launch_warnings = list(bindings.v1LaunchWarning)
+            warnings = [launch_warnings[warning].value for warning in warnings]
+    except json.decoder.JSONDecodeError:
```

Comment:
A few things here:

- When you use a try/except, make your try clause as small as possible.  Put the other stuff in an `else` statement.  Basically, you want to be clear about what code might raise an exception:
  ```python
  try:
    j = r.json()
  except json.decoder.JSONDecodeError:
    warnings = None  # here instead of the top of the function
  else:
    warnings = j.get("warnings")
    if warnings is not None:
      warnings = [bindings.v1LaunchWarning(w) for w in warnings]
  ```

- In this particular case, why even bother with the `r.json()` failure?  We'll see other stack traces when that happens anyway (`req.post()` would fail).

Reply:
updated